wgsl/index.bs

```diff
@@ -3142,36 +3108,451 @@ TODO: deduplicate these tables
+    <td class="nowrap">`frexp(`|e1|`:` |T| `, `|e2|`:` ptr<|I|> `) -> ` |T|
+    <td>(GLSLstd450Frexp)
+  <tr algorithm="vector case, frexp">
+    <td>|T| is f32<br>
+        |I| is i32 or u32
+    <td class="nowrap">`frexp(`|e1|`:` vec|N|<|T|> `, `|e2|`:` ptr&lt;vec|N|&lt;|I|&gt;&gt;`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Frexp)
+  <tr algorithm="scalar case, inverseSqrt">
+    <td>|T| is f32
+    <td class="nowrap">`inverseSqrt(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450InverseSqrt)
+  <tr algorithm="vector case, inverseSqrt">
+    <td>|T| is f32
+    <td class="nowrap">`inverseSqrt(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450InverseSqrt)
+  <tr algorithm="scalar case, ldexp">
+    <td>|T| is f32<br>
+        |I| is i32 or u32
+    <td class="nowrap">`ldexp(`|e1|`:` |T| `, `|e2|`:` ptr<|I|> `) -> ` |T|
+    <td>(GLSLstd450Ldexp)
+  <tr algorithm="vector case, ldexp">
+    <td>|T| is f32<br>
+        |I| is i32 or u32
+    <td class="nowrap">`ldexp(`|e1|`:` vec|N|<|T|> `, `|e2|`:` ptr&lt;vec|N|&lt;|I|&gt;&gt;`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Ldexp)
+  <tr algorithm="scalar case, length">
+    <td>|T| is f32
+    <td class="nowrap">`length(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Length)
+  <tr algorithm="vector case, length">
+    <td>|T| is f32
+    <td class="nowrap">`length(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Length)
+  <tr algorithm="scalar case, log">
+    <td>|T| is f32
+    <td class="nowrap">`log(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Log)
+  <tr algorithm="vector case, log">
+    <td>|T| is f32
+    <td class="nowrap">`log(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Log)
+  <tr algorithm="scalar case, log2">
+    <td>|T| is f32
+    <td class="nowrap">`log2(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Log2)
+  <tr algorithm="vector case, log2">
+    <td>|T| is f32
+    <td class="nowrap">`log2(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Log2)
+  <tr algorithm="scalar case, max">
+    <td>|T| is f32
+    <td class="nowrap">`max(`|e1|`:` |T| `, `|e2|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450FMax)
+  <tr algorithm="vector case, max">
+    <td>|T| is f32
+    <td class="nowrap">`max(`|e1|`:` vec|N|<|T|> `, `|e2|`:` vec|N|<|T|>`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450FMax)
+  <tr algorithm="scalar case, min">
+    <td>|T| is f32
+    <td class="nowrap">`min(`|e1|`:` |T| `, `|e2|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450FMin)
+  <tr algorithm="vector case, min">
+    <td>|T| is f32
+    <td class="nowrap">`min(`|e1|`:` vec|N|<|T|> `, `|e2|`:` vec|N|<|T|>`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450FMin)
+  <tr algorithm="scalar case, mix">
+    <td>|T| is f32
+    <td class="nowrap">`mix(`|e1|`:` |T| `, `|e2|`:` |T| `, `|e3|`:` |T|`) -> ` |T|
+    <td>(GLSLstd450FMix)
+  <tr algorithm="vector case, mix">
+    <td>|T| is f32
+    <td class="nowrap">`mix(`|e1|`:` vec|N|<|T|> `, `|e2|`:` vec|N|<|T|>`, `|e3|`:` vec|N|<|T|>`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Modf)
```

Comment:
There is also `ModfStruct` in GLSL.std.450 which returns a struct instead of using an out param. We could do that and emulate the struct creation on MSL and HLSL if desired.